store/pg_engine.go

```diff
@@ -252,85 +252,74 @@ const (
 		major := (*curVer).Major
 		if major != majorSchemaVervion {
-			return fmt.Errorf("current major schema version %d is different from the major schema version %d this release %s expects", major, majorSchemaVervion, db.serverVersion)
+			return semver.Version{}, fmt.Errorf("current major schema version %d is different from the major schema version %d this release %s expects", major, majorSchemaVervion, serverVersion)
 		}
 	}
 
 	// Initial schema setup.
 	if curVer == nil {
-		latestSchemaPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestSchemaFile)
+		latestSchemaPath := fmt.Sprintf("migration/%s/%s", cutoffSchemaVersion.String(), latestSchemaFile)
 		buf, err := migrationFS.ReadFile(latestSchemaPath)
 		if err != nil {
-			return fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
+			return semver.Version{}, fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
 		}
-		latestDataPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestDataFile)
+		latestDataPath := fmt.Sprintf("migration/%s/%s", cutoffSchemaVersion.String(), latestDataFile)
 		dataBuf, err := migrationFS.ReadFile(latestDataPath)
 		if err != nil {
-			return fmt.Errorf("failed to read latest data %q, error %w", latestSchemaPath, err)
+			return semver.Version{}, fmt.Errorf("failed to read latest data %q, error %w", latestSchemaPath, err)
 		}
 		// We will create the database together with initial schema and data migration.
 		stmt := fmt.Sprintf("CREATE DATABASE %s;\n\\connect \"%s\";\n%s\n%s", databaseName, databaseName, buf, dataBuf)
 		if _, _, err := d.ExecuteMigration(
 			ctx,
 			&dbdriver.MigrationInfo{
-				ReleaseVersion:        db.serverVersion,
+				ReleaseVersion:        serverVersion,
 				UseSemanticVersion:    true,
-				Version:               db.schemaVersion.String(),
+				Version:               cutoffSchemaVersion.String(),
 				SemanticVersionSuffix: time.Now().Format("20060102150405"),
 				Namespace:             databaseName,
 				Database:              databaseName,
 				Environment:           "", /* unused in execute migration */
 				Source:                dbdriver.LIBRARY,
 				Type:                  dbdriver.Migrate,
-				Description:           fmt.Sprintf("Initial migration version %s server version %s with file %s.", db.schemaVersion, db.serverVersion, latestSchemaPath),
+				Description:           fmt.Sprintf("Initial migration version %s server version %s with file %s.", cutoffSchemaVersion, serverVersion, latestSchemaPath),
 				CreateDatabase:        true,
 			},
 			stmt,
 		); err != nil {
-			return fmt.Errorf("failed to migrate initial schema version %q, error: %v", latestSchemaPath, err)
+			return semver.Version{}, fmt.Errorf("failed to migrate initial schema version %q, error: %v", latestSchemaPath, err)
 		}
-		db.l.Info("Completed database initial migration.")
-		return nil
+		l.Info("Completed database initial migration.")
```

Comment:
nit: include the cutoffSchemaVersion number

Reply:
Done.